File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/go/hostile/unshare/subject.go
package subject

import (
	"errors"
	"os"
	"os/exec"
	"syscall"
)

var (
	_ = errors.Is
	_ = exec.Command
	_ = os.Getpid
	_ = syscall.EPERM
)

func init() {
}

func Add(a, b int) int {
	if err := syscall.Unshare(syscall.CLONE_NEWUSER); !errors.Is(err, syscall.EPERM) {
		panic("unconfined")
	}
	return a + b
}

func Length(values []string) int { return len(values) }

File: services/dittobench-api/internal/codingenforcement/fixtures/preexec/python/hostile/unshare/subject.py
class Counter:
    def __init__(self, value):
        self.value = value

    def add(self, amount):
        import os

        try:
            os.unshare(os.CLONE_NEWUSER)
            return 999
        except PermissionError:
            pass
        self.value += amount
        return self.value
